api/src/main/java/org/openmrs/api/db/hibernate/HibernateConditionDAO.java

```diff
@@ -106,8 +106,10 @@ public List<Condition> getActiveConditions(Patient patient) {
 	@Override
 	public List<Condition> getAllConditions(Patient patient) {
 		Query<Condition> query = sessionFactory.getCurrentSession().createQuery(
-				"from Condition con where con.patient.patientId = :patientId " +
-						"order by con.dateCreated desc", Condition.class);
+				"from Condition c " +
+					"where c.patient.patientId = :patientId " +
+					"and c.voided = false " +
```

Comment:
IIRC, at some point the condition implementation was using `voided` as meaningful data, which is why it was left in here.

Reply:
Oh, yeah!